src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -2434,9 +2434,15 @@ public static string Mode(PSObject instance)
 
         private static bool WinCreateSymbolicLink(string path, string strTargetPath, bool isDirectory)
         {
-            int created = NativeMethods.CreateSymbolicLink(path, strTargetPath, (isDirectory ? 1 : 0));
-            bool success = (created == 1) ? true : false;
-            return success;
+            // The new AllowUnprivilegedCreate is only available on Win10 build 14972 or newer
+            var flags = isDirectory ? NativeMethods.SymbolicLinkFlags.Directory : NativeMethods.SymbolicLinkFlags.File;
+            if (Environment.OSVersion.Version.Major >= 10 && Environment.OSVersion.Version.Build >= 14972)
```

Comment:
This should probably be:

```c#
if (Environment.OSVersion.Version.Major == 10 && Environment.OSVersion.Version.Build >= 14972
    || Environment.OSVersion.Version.Major > 10)
```

Reply:
Yeah, we have to cover our bases in case **Windows&nbsp;11** or higher gets released and resets the build number.